nixos/modules/services/security/opensnitch.nix

```diff
@@ -0,0 +1,174 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.opensnitch;
+  inherit (pkgs) opensnitchd opensnitch-ui writeText iptables;
+  inherit (lib) nameValuePair mapAttrsToList;
+  inherit (builtins) listToAttrs replaceStrings concatMap;
+  escapeRuleName = name:
+    replaceStrings [ "*" "." "/" ] [ "all" "-" "-" ] name;
+
+  makeAlwaysRuleFile =  rulePath: name': rule: let name = escapeRuleName name'; in
+    nameValuePair (rulePath + "/${name}.json") {
+      source = writeText name (builtins.toJSON ({
+        inherit name;
+        enabled = true;
+        duration = "always";
+      } // rule));
+    };
+
+  makeAlwaysOperatorFile = rulePath: name: operator:
+    makeAlwaysRuleFile rulePath name {action = "allow"; inherit operator; };
+
+  makePackageRuleFile = rulePath: pkg: let
+    name = "nixos-allow-pkg-${lib.strings.getName pkg}";
+    in makeAlwaysOperatorFile rulePath name {
+          type = "regexp";
+          operand = "process.path";
+          data = "${pkg}/.*";
+    };
+
+  makeHostRuleFile = rulePath: type: value: let
+    name = "nixos-allow-${type}-${value}";
+    in makeAlwaysOperatorFile rulePath name {
+      type = "regexp";
+      operand = "dest.${type}";
+      data = replaceStrings [ "." "*" ] [ "\\." ".*" ] value;
+    };
+
+in
+
+{
+
+  options = {
+    services.opensnitch = with lib; {
+      enable = mkEnableOption "opensnitch";
+
+      uiConfig = mkOption {
```

Comment:
I'll not sure but according to RFC42 convention maybe we want to call this option `settings`.

Reply:
I agree this naming might be not the best. But this config is not for the main opensnitchd daemon but is used by the client application opensnitch-ui ...